test/jdk/java/net/httpclient/EmptyAuthenticate.java

```diff
@@ -24,55 +24,118 @@
+    @MethodSource("args")
+    void test(Version version, boolean secure) throws Exception {
+        HttpTestServer server = createStartedServerRespondingWithEmptyWwwAuthHeader(version, secure);
+        try (HttpClient client = createClient(version, secure)) {
+            HttpRequest request = createRequest(server, secure);
+            HttpResponse<Void> response = client.send(request, HttpResponse.BodyHandlers.discarding());
+            HttpHeaders responseHeaders = response.headers();
+            assertEquals(
+                    "",
+                    responseHeaders.firstValue(WWW_AUTH_HEADER_NAME).orElse(null),
+                    () -> "was expecting empty `%s` header in: %s".formatted(
+                            WWW_AUTH_HEADER_NAME, responseHeaders.map()));
         } finally {
-            server.stop(0);
+            server.stop();
         }
     }
+
+    static Stream<Arguments> args() {
+        return Stream
+                .of(Version.HTTP_1_1, Version.HTTP_2)
+                .flatMap(version -> Stream
+                        .of(true, false)
+                        .map(secure -> Arguments.of(version, secure)));
+    }
+
+    private static HttpTestServer createStartedServerRespondingWithEmptyWwwAuthHeader(Version version, boolean secure)
+            throws IOException {
+        HttpTestServer server = secure
+                ? HttpTestServer.create(version, SSL_CONTEXT)
+                : HttpTestServer.create(version);
+        HttpTestHandler handler = new ServerHandlerRespondingWithEmptyWwwAuthHeader();
+        server.addHandler(handler, "/");
+        server.start();
+        return server;
+    }
+
+    private static final class ServerHandlerRespondingWithEmptyWwwAuthHeader implements HttpTestHandler {
+
+        private int responseIndex = 0;
+
+        @Override
+        public synchronized void handle(HttpServerAdapters.HttpTestExchange exchange) throws IOException {
+            try (exchange) {
+                exchange.getResponseHeaders().addHeader(WWW_AUTH_HEADER_NAME, "");
+                byte[] responseBodyBytes = "test body %d"
+                        .formatted(responseIndex)
+                        .getBytes(StandardCharsets.US_ASCII);
+                exchange.sendResponseHeaders(401, responseBodyBytes.length);
+                exchange.getResponseBody().write(responseBodyBytes);
+            } finally {
+                responseIndex++;
+            }
+        }
+
+    }
+
+    private static HttpClient createClient(Version version, boolean secure) {
+        HttpClient.Builder clientBuilder = HttpClient.newBuilder().version(version).proxy(NO_PROXY);
+        if (secure) {
+            clientBuilder.sslContext(SSL_CONTEXT);
+        }
+        return clientBuilder.build();
+    }
+
+    private static HttpRequest createRequest(HttpTestServer server, boolean secure) {
+        URI uri = URI.create("%s://%s/".formatted(secure ? "https" : "http", server.serverAuthority()));
```

Comment:
```suggestion
        URI uri = URI.create("%s://%s/EmptyAuthenticate/%s".formatted(secure ? "https" : "http", server.serverAuthority(), server.gerVersion()));
```

Reply:
Applied it in 7b5c849002c2302a3b4d4cdec3fd427fc9a1c4f7.